Tests/AK/TestAKMath.cpp

```diff
@@ -0,0 +1,2305 @@
+    { 982.4999999999495, 5.780571000783179e+295 },
+    { 983.3999999999494, 1.0786926907302273e+296 },
+    { 984.2999999999493, 2.012911736361634e+296 },
+    { 985.1999999999493, 3.7562261181543264e+296 },
+    { 986.0999999999494, 7.009365783820884e+296 },
+    { 986.9999999999493, 1.3079939052107038e+297 },
+    { 987.8999999999492, 2.440802932581078e+297 },
+    { 988.7999999999493, 4.554699323876158e+297 },
+    { 989.6999999999493, 8.499369471414227e+297 },
+    { 990.5999999999492, 1.586038424817896e+298 },
+    { 991.4999999999491, 2.959652352400055e+298 },
+    { 992.3999999999492, 5.522906576537894e+298 },
+    { 993.2999999999493, 1.0306108090171567e+299 },
+    { 994.1999999999491, 1.923187772494712e+299 },
+    { 995.099999999949, 3.588795281315161e+299 },
+    { 995.9999999999491, 6.696928794677748e+299 },
+    { 996.8999999999492, 1.2496911014815119e+300 },
+    { 997.799999999949, 2.3320060538242254e+300 },
+    { 998.6999999999489, 4.351677169362713e+300 },
+    { 999.599999999949, 8.120516735066348e+300 },
+    { 1000.4999999999491, 1.5153420044288281e+301 },
+    { 1001.3999999999489, 2.8277281671869557e+301 },
+    { 1002.2999999999488, 5.276727342166179e+301 },
+    { 1003.1999999999489, 9.846721395171373e+301 },
+    { 1004.099999999949, 1.8374631840333623e+302 },
+    { 1004.9999999999488, 3.4288275428744666e+302 },
+    { 1005.8999999999487, 6.398418439583324e+302 },
+    { 1006.7999999999488, 1.1939870995578152e+303 },
+    { 1007.6999999999489, 2.228058710713709e+303 },
+    { 1008.5999999999487, 4.1577045683533146e+303 },
+    { 1009.4999999999486, 7.758551062673154e+303 },
+    { 1010.3999999999487, 1.447796821599493e+304 },
+    { 1011.2999999999488, 2.7016843991890834e+304 },
+    { 1012.1999999999487, 5.041521354326949e+304 },
+    { 1013.0999999999485, 9.40781150224785e+304 },
+    { 1013.9999999999486, 1.7555597019514502e+305 },
+    { 1014.8999999999487, 3.275990241066662e+305 },
+    { 1015.7999999999486, 6.11321394973505e+305 },
+    { 1016.6999999999484, 1.1407660598850594e+306 },
+    { 1017.5999999999485, 2.1287447389965618e+306 },
+    { 1018.4999999999486, 3.972378144088655e+306 },
+    { 1019.3999999999485, 7.412719726588237e+306 },
+};
+
+TEST_CASE(pow)
+{
+    EXPECT_EQ(AK::pow(10., 0.), 1.);
+    EXPECT_EQ(AK::pow(10., 1.), 10.);
+    EXPECT_EQ(AK::pow(10., 2.), 100.);
+    EXPECT_EQ(AK::pow(10., 3.), 1'000.);
+    EXPECT_EQ(AK::pow(10., 4.), 10'000.);
+    EXPECT_EQ(AK::pow(10., 5.), 100'000.);
+    EXPECT_EQ(AK::pow(10., 6.), 1'000'000.);
+    EXPECT_EQ(AK::pow(10., 19.), 10'000'000'000'000'000'000.);
+    // Should not take the integer fast path anymore, since we're beyond 64-bit integers.
+    EXPECT_EQ(AK::pow(10., 20.), 100'000'000'000'000'000'000.);
+    EXPECT_EQ(AK::pow(10., 21.), 1'000'000'000'000'000'000'000.);
+    EXPECT_EQ(AK::pow(10., 22.), 10'000'000'000'000'000'000'000.);
+    EXPECT_EQ(AK::pow(10., 23.), 100'000'000'000'000'000'000'000.);
+}
```

Comment:
So when I run this test case on `master`, it succeeds. Can you add a test that fails on `master` and succeeds after your PR?

Reply:
Huh. That's clearly UB though.

Reply:
My test case was in Maps by the way, so I don't have numbers unfortunately